py-server/history_tutor/history_tutor.py

```diff
@@ -136,50 +136,47 @@ def __init__(self):
 
         self.lesson_context = load_file(BLACK_DEATH_TUTOR_CONTEXT)
 
-    def chat(self, message_history, world_state, message):
-        if not world_state:
-            print("No world state received, creating new one")
-            world_state = self.build_world_state()
-            print("World state created", world_state)
-        else:
-            world_state = self.update_world_state(world_state, message_history, message)
+    def chat(self, message: str):
+        response = self.chat_model.invoke(self._state.message_history)
+        return response.content
 
+    def _build_message_history(self, state):
         system_prompt = PromptTemplate.from_template(PROMPT).format(
             lesson_context=self.lesson_context,
-            world_state=json.dumps(world_state),
+            world_state=json.dumps(state.world_state),
         )
         messages = [SystemMessage(system_prompt)]
-        messages.extend(message_history)
-        messages.append(HumanMessage(message))
-        response = self.chat_model.invoke(messages)
-        return response.content, world_state
+        messages.extend(state.message_history)
+        messages.append(HumanMessage(state.message))
+        return messages
 
-    def build_world_state(self):
+    def _init_world_state(self):
+        # TODO(alalama): update world state to a dict
         if USE_FAKE_WORLD_STATE:
```

Comment:
no need to keep the FAKE_WORLD_STATE that was just me hacking around. Can remove it for now & if we need something more structured as a dev tool we can add it later.

Reply:
done in fa40e6f.